integration_tests/suite/test_phonebook_http.py

```diff
@@ -358,3 +391,304 @@ def test_get_multi_tenant(self, sub, main):
+                    'lastname': 'McContact',
+                    'number': str(1000000000 + i),
+                }
+                for i in range(5000)
+            ],
+        )
+        assert not errors
+        assert len(cls.contacts) == 5000
+
+        source_body = {
+            # 'auth': {'host': 'auth', 'port': 9497, 'prefix': None, 'https': False},
+            'first_matched_columns': ['number', 'firstname', 'lastname'],
+            'format_columns': {
+                'phone': '{number}',
+                'reverse': '{firstname} {lastname}',
+            },
+            'name': 'phonebook-lookup-test',
+            'searched_columns': ["firstname", "lastname", "number"],
+            'phonebook_uuid': phonebook['uuid'],
+        }
+
+        source = cls.stack.enter_context(phonebook_source(client, source_body))
+
+        display_body = {
+            'name': 'default',
+            'columns': [
+                {'title': 'firstname', 'field': 'firstname'},
+                {'title': 'lastname', 'field': 'lastname'},
+                {'title': 'number', 'field': 'number'},
+            ],
+        }
+        display = client.displays.create(display_body)
+
+        profile_body = {
+            'name': 'default',
+            'display': display,
+            'services': {
+                'lookup': {'sources': [source]},
+                'reverse': {'sources': [source]},
+                'favorites': {'sources': [source]},
+            },
+        }
+        profile = client.profiles.create(profile_body)
+
+        cls.source_uuid = source['uuid']
+        cls.source_name = source['name']
+        cls.display_uuid = display['uuid']
+        cls.profile_uuid = profile['uuid']
+        cls.stack.callback(client.displays.delete, cls.display_uuid)
+        cls.stack.callback(client.profiles.delete, cls.profile_uuid)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.stack.close()
+        super().tearDownClass()
+
+    def test_plugin_lookup(self):
+        result, timing = timed(
+            self.client.directories.lookup, term=' 5', profile='default'
+        )
+        expected_count = 111
+
+        assert_that(
+            result,
+            has_entries(
+                results=all_of(
+                    has_length(expected_count),
+                    has_item(
+                        has_entries(
+                            backend='phonebook',
+                            source=self.source_name,
+                            column_values=contains(
+                                'Contact 5',
+                                'McContact',
+                                '1000000005',
+                            ),
+                        )
+                    ),
+                ),
+            ),
+        )
+
+        # bound response time to a linear function of the number of contacts
+        assert_that(timing['elapsed'] < len(self.contacts) * 0.01)
```

Comment:
Please use `assert` for simple assertions, and add an assertion message.

Also, this places a threshold at 5000 * 0.01 = 50 seconds, which is unacceptable performance IMHO.

Reply:
Add fix ceiling